python/ray/util/client/dataclient.py

```diff
@@ -51,6 +51,8 @@ def __init__(self, client_worker: "Worker", client_id: str, metadata: list):
         self.cv = threading.Condition(lock=self.lock)
 
         self.request_queue = queue.Queue()
+        # This is to avoid deadlock in gc and the code.
+        self.gc_queue = queue.Queue()
```

Comment:
If we use a deque here, it may solve https://github.com/ray-project/ray/issues/22082.